core/cap-0046-07.md

```diff
@@ -506,13 +483,14 @@ The model retained in this CAP is a flat rate per byte model for simplicity. It
 
 Transactions need to be propagated to peers on the network.
 
-Fees to propagate transactions as they are now can just be included in the "Historical storage" cost (that already charges for the cost of `TransactionEnvelope`).
+At the networking layer, transactions compete for bandwidth on a per ledger basis (`ledgerMaxPropagateSizeBytes`).
+A minimum is also defined as spam countermeasure.
 
-In order to support payloads that get discarded by validators, bandwidth needs to be tracked separately.
+Note that validators may apply additional market dynamics due to implementation constraints, especially when trying to balance propagating large transactions vs smaller ones. See [CAP-0042: Multi-Part Transaction Sets](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0042.md).
 
-Transaction contains an optional `payload` (Hash + size), and the `TransactionEnvelope` gets the actual payload, that gets cleared before applying transactions.
+##### Ephemeral payload
```

Comment:
I'm not sure what this means and what are the proofs of availability. FWIW any payload that only exists for the lifespan of a function invocation doesn't need to be declared explicitly, as it would just be paid for via the regular budget metering mechanisms.

Reply:
We need to talk about this outside of this PR. I originally put it there so that we have the right extension points and concepts when we design fees. We do not need to ship this in v1. TLDR; is that contracts have access to `ephemeralPayload` but the actual binary blob that corresponds to that payload is not stored anywhere by default, only propagated. This allows to implement efficient rollup protocols. I added a link to an article about.

Reply:
Sounds good, it just seemed weird to include that without any context.